### ISSUE TITLE
Add check for projects to have a Changelog

### ISSUE THREAD

**Issue by tomschr** [fonttools/fontspector#462]
First, thank you for all the efforts with the fonts! Highly appreciated! :+1:

## Problem
Currently, your project is missing a Changelog.

## Benefits

To make it easier for users, packagers, and contributors to see precisely what notable changes have been made between each release (or version) of the project.

## Solution

I guess, you know how to write a Changelog. :wink:  In case you need some inspirations, refer to the following links:

* https://www.itsupportguides.com/blog/how-to-write-changelog-and-why-it-matters/
* https://keepachangelog.com/

Thanks!

**Comment by davelab6** [comment]
@tomschr I agree, font projects should indeed have changelogs, and we should check for one in all Google Fonts upstream projects; therefore I'm moving this to the fontbakery project issue tracker.

**Comment by felipesanches** [comment]
All of Font Bakery's `"new check proposal"` issues are being either closed or migrated to the [Fontspector issue tracker](https://github.com/fonttools/fontspector/issues), as there's no intention to implement new checks on the Font Bakery codebase anymore.

If you believe this proposal is still valid, please post a message here asking for it to be migrated. Otherwise, on **July 20, 2025** (30 days from now) it will be closed due to inactivity.

**Comment by felipesanches** [comment]
The fontspector issue tracker does not currently use priority/importance labels, but when it was hosted at the fontbakery issue tracker:

* @felipesanches applied "P3 Soon" on Dec 21, 2018

**Comment by simoncozens** [comment]
The original proposal was about Fontbakery/Fontspector having a changelog, which it certainly has! Currently fontspector is focusing on binary outputs and does not check the build environment - maybe in the future it will have a "repo profile", but since nobody asked for one since the issue was deprecated in June, I'm closing this.

**Comment by felipesanches** [comment]
> The original proposal was about Fontbakery/Fontspector having a changelog, which it certainly has!

No it wasn't. It was about font projects (upstream repos with font sources) to have a Changelog listing what which improvements were made on the font family since its last release.

While this sounds great from a purely software development workflow, it is a bit less common to see font designers doing so. So a check for this would be an attempt to shift the culture towards something we believe is better. But it would also probably be a bit annoying to the designers who still do not have that development habit.


> maybe in the future it will have a "repo profile"

As I mentioned at https://github.com/fonttools/fontspector/issues/465#issuecomment-3346962605 it would be better to have this as checks in the google fonts profile instead of creating such a specific-purpose profile.

I believe that instead of grouping by checking aspect, the Fontbakery/Fontspector profiles should always be defined at the boundaries of vendor needs.

Yes, I agree to keep this issue closed. My comment here was just for clarification purposes.

**Comment by felipesanches** [comment]
Oh, sorry! Indeed the original proposal was about fontbakery but @davelab6's followup was about font projects.

**Comment by tomschr** [comment]
Thanks Felipe for the addition. 👍 

> While this sounds great from a purely software development workflow, it is a bit less common to see font designers doing so. [...] But it would also probably be a bit annoying to the designers who still do not have that development habit.

That's true and see this omission in many fonts. It may be a bit unusual for font designers, but some projects do that and understand its importance. Do great things and talk about them!

In my humble opinion, it's also in the interest of font designers. I would argue that when font designers are working on different fonts, they themselves no longer know exactly what they changed last year. A changelog would solve that.

There are so many great fonts out there. How do we all know what's new when nobody tells us? They don't have to write novels. When they release a new version it's not a big deal to write one or two sentences. 🙂

Anyway, thanks for listening.
